src/oic/utils/rp/__init__.py

```diff
@@ -110,6 +110,47 @@ def _err(self, txt):
         logger.error(sanitize(txt))
         raise OIDCError(txt)
 
+    def _do_code(self, response, authresp):
+        """Perform code flow."""
+        # get the access token
+        try:
+            args = {
+                "code": authresp["code"],
+                "redirect_uri": self.registration_response["redirect_uris"][0],
+                "client_id": self.client_id,
+                "client_secret": self.client_secret,
+            }
+
+            try:
+                args["scope"] = response["scope"]
+            except KeyError:
+                pass
+
+            atresp = self.do_access_token_request(
+                state=authresp["state"],
+                request_args=args,
+                authn_method=self.registration_response[
+                    "token_endpoint_auth_method"
+                ],
+            )
+            msg = "Access token response: {}"
+            logger.info(msg.format(sanitize(atresp)))
+        except Exception as err:
+            logger.error("%s" % err)
+            raise
+
+        if isinstance(atresp, ErrorResponse):
+            msg = "Error response: {}"
+            self._err(msg.format(sanitize(atresp.to_dict())))
+
+        _token = atresp["access_token"]
+
+        try:
+            _id_token = atresp["id_token"]
+        except KeyError:
+            pass
+        return _token, _id_token
```

Comment:
_id_token is undefined if the try except block gets invoked with a KeyError.
The default was setup in the other method this has been extracted from, so returning None and dealing with it in the caller seems okay.

Reply:
Changed the code a little bit so it doesn't overwrite the existing token if it existed.